src/components/legend.ts

```diff
@@ -144,13 +164,17 @@ export module Components {
 
       let availableWidthForEntries = Math.max(0, (availableWidth - this._padding));
 
-      let entryNames = this._colorScale.domain().slice();
-      entryNames.sort(this.comparator());
+      let entryLabels: d3.Map<string> = d3.map<string>();
+      this._colorScale.domain().slice().forEach((d) => {
+        entryLabels.set(this._formatter(d), d);
+      });
+      let entryNames = entryLabels.keys();
+      entryNames = entryNames.sort(this.comparator()).map((d) => entryLabels.get(d));
```

Comment:
I want to sort the labels by `this.formatter(entryID)` instead of `entryID`.  Is there any good way to do something like `entryNames.sort(this.comparator(this.formatter()))`?